cmd/yggdrasil/main.go

```diff
@@ -271,15 +272,53 @@ func main() {
 			n.tun.SetupAdminHandlers(n.admin)
 		}
 	}
+	
+	//Windows service shutdown service
+	minwinsvc.SetOnExit(func() {
+		logger.Infof("Shutting down service ...")
+		sigCh <- os.Interrupt
+		// Wait for all parts to shutdown properly
+		<-doneCh
```

Comment:
Is this the part (blocking the exit handler) specifically that fixes the problem? If so, I'm OK with keeping a done channel but then:
1. It doesn't need to be a global, it can just be defined further up in `main`
2. You don't need to replace all of the context code either, you can just do something like this in `main`:
```
done := make(chan struct{})
defer close(done)
```
... and then change `SetOnExit` to:
```
minwinsvc.SetOnExit(func() {
  cancel()
  <-done
})
```
... which would be a far less invasive set of changes.

Reply:
Okay, I'll try to simplify it :)